photon-ml/src/test/scala/com/linkedin/photon/ml/model/CoefficientsTest.scala

```diff
@@ -66,7 +81,77 @@ class CoefficientsTest {
       MathConst.HIGH_PRECISION_TOLERANCE_THRESHOLD)
   }
 
+  /**
+   * Test that means and variances use the same dynamic type for Vector
+   */
+  @Test
+  def testPreconditions1(): Unit = {
+
+    val d: Vector[Double] = new DenseVector[Double](Array[Double](1, 2, 3))
+    val s: Vector[Double] = new SparseVector[Double](Array[Int](0, 2), Array[Double](0, 3), 3)
```

Comment:
Typically for cases like this, we'd use a DataProvider for these, and then the test would declare the expectedException.